molpipeline/mol2mol/filter.py

```diff
@@ -120,38 +185,318 @@ def pretransform_single(self, value: RDKitMol) -> OptionalMol:
+        keep: bool, optional (default: True)
+            If True, molecules containing the specified descriptors are kept, else removed.
+        mode: Literal["any", "all"], optional (default: "any")
+            If "any", at least one of the specified descriptors must be present in the molecule.
+            If "all", all of the specified descriptors must be present in the molecule.
+        name: Optional[str], optional (default: None)
             Name of the pipeline element.
         n_jobs: int, optional (default: 1)
             Number of parallel jobs to use.
         uuid: str, optional (default: None)
             Unique identifier of the pipeline element.
         """
         super().__init__(name=name, n_jobs=n_jobs, uuid=uuid)
+        self.descriptors = descriptors
+        self.keep = keep
+        self.mode = mode
+
+    @property
+    def descriptors(self) -> dict[str, tuple[Optional[float], Optional[float]]]:
+        """Get allowed descriptors as dict."""
+        return self._descriptors
+
+    @descriptors.setter
+    def descriptors(
+        self, descriptors: dict[str, tuple[Optional[float], Optional[float]]]
+    ) -> None:
+        """Set allowed descriptors as dict.
+
+        Parameters
+        ----------
+        descriptors: dict[str, tuple[Optional[float], Optional[float]]]
+            Dictionary of RDKit descriptors to filter by.
+        """
+        self._descriptors = descriptors
+        if not all(hasattr(Descriptors, descriptor) for descriptor in descriptors):
+            raise ValueError(
+                "You are trying to use an invalid descriptor. Use RDKit Descriptors module."
+            )
+
+    def get_params(self, deep: bool = True) -> dict[str, Any]:
+        """Get parameters of DescriptorFilter.
+
+        Parameters
+        ----------
+        deep: bool, optional (default: True)
+            If True, return the parameters of all subobjects that are PipelineElements.
+
+        Returns
+        -------
+        dict[str, Any]
+            Parameters of DescriptorFilter.
+        """
+        params = super().get_params(deep=deep)
+        if deep:
+            params["descriptors"] = {
+                descriptor: (count_tuple[0], count_tuple[1])
+                for descriptor, count_tuple in self.descriptors.items()
+            }
+        else:
+            params["descriptors"] = self.descriptors
+        params["keep"] = self.keep
+        params["mode"] = self.mode
+        return params
+
+    def pretransform_single(self, value: RDKitMol) -> OptionalMol:
+        """Invalidate or validate molecule based on specified RDKit descriptors.
+
+        Parameters
+        ----------
+        value: RDKitMol
+            Molecule to check.
+
+        Returns
+        -------
+        OptionalMol
+            Molecule that matches defined descriptors filter, else InvalidInstance.
+        """
+        for descriptor, (min_count, max_count) in self.descriptors.items():
```

Comment:
I think I have an Idea how to improve the logic here. Will make PR to this branch tomorrow.